### ISSUE TITLE
Jetpack periodically DDoSes our own site and brings down mysql to its knees

### ISSUE THREAD

**Issue by archon810** [Automattic/jetpack#19823]
#### Steps to reproduce the issue

Periodically, and I do note quite rarely (every couple of months), a busy site of ours with several thousand concurrent visitors goes down due to the database being overloaded with queries that all hang and wait for each other to finish.

When this happens, `show processlist` shows all the queries belong to Jetpack, and presumably each visitor (not logged in) unnecessarily triggers the same or similar query, which quickly overwhelms the server and in some cases even breaks replication.

Here's an example query:

```
UPDATE `wp_options` SET `option_value` = 'a:9:{s:32:\"XXXXXXXXXXXXXXXXXXXXXXXX\";a:15:{s:4:\"name\";s:20:\"Custom content types\";s:11:\"description\";s:74:\"Display different types of content on your site with custom content types.\";s:4:\"sort\";s:2:\"34\";
...
 WHERE `option_name` = '_transient_jetpack_file_data_9.6'
```

The queries are really long but they all attempt to update `_transient_jetpack_file_data_9.6` (or whatever the installed version of Jetpack is).

Comparing some of them, they're also not the same and seem to differ quite a bit. I am not sure if any of the data contains private tokens, so I'm not sharing it here (I saved it in a gist, available upon request).

When I ran into this situation this morning, there were hundreds of these running continuously and locked up mysql. I didn't save the full log to see if there are other types of Jetpack queries in there, but safe to say if I did, they would have been all transient jetpack queries.

Unfortunately, I don't have reproduction steps as this happens randomly, but try to think what would happen if whatever is triggering this query to run doesn't complete for a few seconds, triggering another and another and another with every concurrent visit to the page.

In general, it's a bad idea to trigger potentially long-running tasks from user visits. Ideally, this would happen in wp cron or when an admin (or at least a logged in user) interacts with the page or logs into wp-admin.

I've run into several situations with various plugins that resulted in such concurrency-related DDoSes (and sometimes the end result is the plugin's own settings get reset because of failed db reads), but Jetpack is the only such plugin in the past several years.

Thank you.

**Comment by jeherve** [comment]
Thanks for the report. It's something we've worked on in the past. See previous discussions in #8055 and fix in #8056. 

The transient solution isn't always ideal though, as you've experienced. This was mentioned in #8457. 

I'll close your issue since we already have #8457, but I'll mention your report there so we can keep working on that.

Thanks again for the report.

**Comment by archon810** [comment]
This just happened again. Brought down our sites for hours until someone woke me up. @jeherve Please can Jetpack dedicate some resources to resolve this issue?

![image](https://user-images.githubusercontent.com/469543/121494844-f43ad480-c98d-11eb-962f-ee8e3936f865.png)
![image](https://user-images.githubusercontent.com/469543/121495050-22b8af80-c98e-11eb-83fd-1c8bc8028652.png)
